crypto/crypto.go

```diff
@@ -76,6 +76,14 @@ func CreateAddress(b common.Address, nonce uint64) common.Address {
 	return common.BytesToAddress(Keccak256(data)[12:])
 }
 
+// CreateAddress2 creates an ethereum address given the address bytes, initial
+// contract code and a salt.
+// TODO(rjl493456442) considering address collision, should we add the 0xff as the input prefix?
+func CreateAddress2(b common.Address, salt []byte, code []byte) common.Address {
+	data, _ := rlp.EncodeToBytes([]interface{}{b, salt, code})
+	return common.BytesToAddress(Keccak256(data)[12:])
```

Comment:
I'm unsure whether this method is correct or not. The spec seems to state `sha3(msg.sender ++ salt ++ init_code)[12:]`. Given that `sender`, `salt` and `code` are all bytes, I'd interpret the formula as simple append, not RLP encoding. Might be worth a clarification on the EIP.

Reply:
For the rlp encoding, no matter address is `common.Address` or byte slice, the result should be same.
And RLP or simple Append, yes we should give some feedbacks to EIP and make a clarification

Reply:
https://github.com/ethereum/EIPs/pull/1014#issuecomment-406193182